terraform/files/bin/create_cluster.sh

```diff
@@ -4,6 +4,25 @@
 ## license: Apache-2.0
 
 STARTTIME=$(date +%s)
+
+wait_for_k8s_resource_matching() {
+  local SLEEP=0
+  until kubectl $2 get $1 -o=jsonpath='{.metadata.name}' >/dev/null 2>&1; do
+    echo "[$SLEEP] waiting for $1"
+    sleep 10
+    let SLEEP+=10
+  done
+}
+
+wait_for_k8s_resources_matching() {
+  local SLEEP=0
+  until [ ! -z $(kubectl $3 get $2 --template '{{if len .items}}{{with index .items 0}}{{.metadata.name}}{{end}}{{end}}') ]; do
+    echo "[$SLEEP] waiting for $1"
```

Comment:
Just a UX suggestion, first I thought you wanted to implement some nice back off algorithm. Then I realized how you meant it. I would add some notion to show it is seconds.